src/app/fdctl/run/tiles/fd_gossip.c

```diff
@@ -0,0 +1,623 @@
+  ctx->shred_contact_out_chunk       = ctx->shred_contact_out_chunk0;
+
+  /* Set up repair contact info tile output */
+  fd_topo_link_t * repair_contact_out = &topo->links[ tile->out_link_id[ REPAIR_OUT_IDX ] ];
+  ctx->repair_contact_out_mcache      = repair_contact_out->mcache;
+  ctx->repair_contact_out_sync        = fd_mcache_seq_laddr( ctx->repair_contact_out_mcache );
+  ctx->repair_contact_out_depth       = fd_mcache_depth( ctx->repair_contact_out_mcache );
+  ctx->repair_contact_out_seq         = fd_mcache_seq_query( ctx->repair_contact_out_sync );
+  ctx->repair_contact_out_mem         = topo->workspaces[ topo->objs[ repair_contact_out->dcache_obj_id ].wksp_id ].wksp;
+  ctx->repair_contact_out_chunk0      = fd_dcache_compact_chunk0( ctx->repair_contact_out_mem, repair_contact_out->dcache );
+  ctx->repair_contact_out_wmark       = fd_dcache_compact_wmark ( ctx->repair_contact_out_mem, repair_contact_out->dcache, repair_contact_out->mtu );
+  ctx->repair_contact_out_chunk       = ctx->repair_contact_out_chunk0;
+
+  /* Set up repair contact info tile output */
+  fd_topo_link_t * pack_out = &topo->links[ tile->out_link_id[ PACK_OUT_IDX ] ];
+  ctx->pack_out_mcache      = pack_out->mcache;
+  ctx->pack_out_sync        = fd_mcache_seq_laddr( ctx->pack_out_mcache );
+  ctx->pack_out_depth       = fd_mcache_depth( ctx->pack_out_mcache );
+  ctx->pack_out_seq         = fd_mcache_seq_query( ctx->pack_out_sync );
+  ctx->pack_out_mem         = topo->workspaces[ topo->objs[ pack_out->dcache_obj_id ].wksp_id ].wksp;
+  ctx->pack_out_chunk0      = fd_dcache_compact_chunk0( ctx->pack_out_mem, pack_out->dcache );
+  ctx->pack_out_wmark       = fd_dcache_compact_wmark ( ctx->pack_out_mem, pack_out->dcache, pack_out->mtu );
+  ctx->pack_out_chunk       = ctx->pack_out_chunk0;
+
+  fd_topo_link_t * sign_in  = &topo->links[ tile->in_link_id[ SIGN_IN_IDX ] ];
+  fd_topo_link_t * sign_out = &topo->links[ tile->out_link_id[ SIGN_OUT_IDX ] ];
+  if ( fd_keyguard_client_join( fd_keyguard_client_new( ctx->keyguard_client,
+                                                            sign_out->mcache,
+                                                            sign_out->dcache,
+                                                            sign_in->mcache,
+                                                            sign_in->dcache ) )==NULL ) {
+    FD_LOG_ERR(( "Keyguard join failed" ));
+  }
+
+  /* Valloc setup */
+  void * alloc_shalloc = fd_alloc_new( alloc_shmem, 3UL );
+  if( FD_UNLIKELY( !alloc_shalloc ) ) { 
+    FD_LOG_ERR( ( "fd_allow_new failed" ) ); }
+  fd_alloc_t * alloc = fd_alloc_join( alloc_shalloc, 3UL );
+  if( FD_UNLIKELY( !alloc ) ) {
+    FD_LOG_ERR( ( "fd_alloc_join failed" ) ); 
+  }
+
+  fd_valloc_t valloc = fd_alloc_virtual( alloc );
+
+  /* Gossip set up */
+  ctx->gossip = fd_gossip_join( fd_gossip_new( ctx->gossip, ctx->gossip_seed, valloc ) );
```

Comment:
for a future PR as a TODO: we can remove this valloc quite easily and make gossip bump allocated